gelataio_boy_control/src/scooping.cpp

```diff
@@ -171,9 +172,58 @@ bool ScoopingMain::perform_scoop() {
     return right_arm.moveJoint("wrist_right", -0.5);
 }
 
-bool ScoopingMain::depart_from_scoop() {
-    right_arm.setPlanningTime(5.0);
-    ROS_WARN("Depart from scoop motion not yet implemented");
-    return true;
+bool ScoopingMain::drop_ice() {
+    ROS_WARN("Rotating the wrist_right");
+
+    return right_arm.moveJoint("wrist_right", 0.87);
 }
 
+bool ScoopingMain::depart_from_scoop(geometry_msgs::Point point_above_cup) {
+    right_arm.setPlanningTime(15.0);
+    Pose hold_scoop_pose;
+    hold_scoop_pose.position = point_above_cup;
+    moveit_msgs::JointConstraint wristConstraint;
+    moveit_msgs::Constraints constraints;
+    tf2::Quaternion q_start;
+    double roll = -20;
+    double pitch = 40;
+    double yaw = 45 + 90;
+    bool scooper_hold = false;
+    bool drop_success = false;
+
+    q_start.setRPY(roll/180*M_PI, pitch/180*M_PI, yaw/180*M_PI);
+    hold_scoop_pose.orientation.x = q_start.x();
+    hold_scoop_pose.orientation.y = q_start.y();
+    hold_scoop_pose.orientation.z = q_start.z();
+    hold_scoop_pose.orientation.w = q_start.w();
```

Comment:
You can set it to identity [0, 0, 0, 1]. The orientation is ignored now as we have the dummy ball joint in the scooper.

Reply:
doesn't matter . its ignored anyway